src/module_type_translator.ts

```diff
@@ -377,7 +377,7 @@ export class ModuleTypeTranslator {
    *    function statement; for overloads, name will have been merged.
    */
   getFunctionTypeJSDoc(fnDecls: ts.SignatureDeclaration[], extraTags: jsdoc.Tag[] = []):
-      [jsdoc.Tag[], string[]] {
+      {tags: jsdoc.Tag[], parameterNames: string[], thisReturnType: ts.Type|null} {
```

Comment:
Can you doc what thisReturnType is?  It's the actual return type if the function returns this, right?

Reply:
Good catch, I forgot that we don't annotate arrows. Done.

Reply:
Don.